src/zenml/new/pipelines/pipeline.py

```diff
@@ -739,6 +741,50 @@ def log_pipeline_deployment_metadata(
         except Exception as e:
             logger.debug(f"Logging pipeline deployment metadata failed: {e}")
 
+    def register_running_versions(
+        self, pipeline: PipelineRunResponseModel
+    ) -> None:
+        """Registers the running versions of the models used in the given pipeline run.
+
+        Args:
+            pipeline: The pipeline run response model.
+
+        Raises:
+            KeyError: No running model version found for @step model configs.
```

Comment:
```suggestion
            KeyError: No running model version found for @step `model_config`s.
```

Maybe? Just to minimise confusion.